includes/Checker/Checks.php

```diff
@@ -0,0 +1,182 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\Checks
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Checker\Check_Context;
+use Exception;
+
+/**
+ * Class to run checks on a plugin.
+ *
+ * @since n.e.x.t
+ */
+class Checks implements Preparation {
+
+	/**
+	 * Context for the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 * @var Check_Context
+	 */
+	protected $check_context;
+
+	/**
+	 * Internal flag for whether the environment is prepared.
+	 *
+	 * @since n.e.x.t
+	 * @var bool
+	 */
+	protected $prepared = false;
+
+	/**
+	 * Sets the main context and the main file of the plugin to check.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $plugin_main_file Absolute path to the plugin main file.
+	 */
+	public function __construct( $plugin_main_file ) {
+		$this->check_context = new Check_Context( $plugin_main_file );
+	}
+
+	/**
+	 * Prepares the environment for running checks and returns a cleanup function.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable Cleanup function to revert any changes made here.
+	 *
+	 * @throws Exception Thrown when preparation fails.
+	 */
+	public function prepare() {
+		// TODO: Add preparations once implemented.
+		$this->prepared = true;
+
+		return function() {
+			$this->prepared = false;
+		};
+	}
+
+	/**
+	 * Runs all checks against the plugin.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return Check_Result Object containing all check results.
+	 *
+	 * @throws Exception Thrown when checks fail with critical error.
+	 */
+	public function run_all_checks() {
+		if ( ! $this->prepared ) {
+			throw new Exception(
+				__( 'Environment not prepared to run checks. The Checks::prepare() method must be called first.', 'plugin-check' )
+			);
+		}
+
+		$result = new Check_Result( $this->main_context, $this->check_context );
```

Comment:
This is copied over from the PoC, but doesn't work now. Please double-check, we don't have `$main_context` here. :)